### PR TITLE
Re-organize and catkinize packages

### PR DESCRIPTION
**Pull request by afakihcpr**
Repo consists now of the following catkin packages
gflip_engine: actual gflip matching package
gflip_vocabulary: stuff related to create and work with vocabularies
gflip: meta package

@efernandez  @servos

### CODE REVIEW

**Comment by servos** [comment]
This is just moving packages yes?

If so then LGTM.

**Comment by efernandez** [comment]
Same comment as @servos

**Comment by afakihcpr** [comment]
Yeah it is mostly moving files. The CMakelists have been modified though.

File: gflip_engine/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+TARGET_LINK_LIBRARIES(gflip_cl ${catkin_LIBRARIES} ${PROJECT_NAME})
+
+ADD_EXECUTABLE(gflip_cl_onequery examples/gflip_cl_onequery.cpp)
+TARGET_LINK_LIBRARIES(gflip_cl_onequery ${catkin_LIBRARIES} ${PROJECT_NAME})
+
+
+
+install(
+  TARGETS
+    ${PROJECT_NAME}
```
Comment (efernandez) [inline gflip_engine/CMakeLists.txt]: executables should be installed too

File: gflip_vocabulary/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+include_directories(
+    include
+    ${catkin_INCLUDE_DIRS}
+    ${Boost_INCLUDE_DIRS}
+    ${FLIRT_INCLUDE_DIRS}
+)
+catkin_package(
+  INCLUDE_DIRS include
+  LIBRARIES ${PROJECT_NAME}
+#  CATKIN_DEPENDS
```
Comment (efernandez) [inline gflip_vocabulary/CMakeLists.txt]: remove

File: gflip_vocabulary/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,38 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(gflip_vocabulary)
+
+find_package(catkin REQUIRED)
+find_package(FLIRT REQUIRED)
+
+include_directories(
```
Comment (efernandez) [inline gflip_vocabulary/CMakeLists.txt]: below and add separation line

**Pull request merged**